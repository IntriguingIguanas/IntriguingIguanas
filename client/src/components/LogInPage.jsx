import React from 'react'
import { connect } from 'react-redux'
import { LogIn } from '../actions/LogIn'

export class LogInPage extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      username: '',
      password: ''
    }
  
    this.handleUsernameChange = this.handleUsernameChange.bind(this)
    this.handlePasswordChange = this.handlePasswordChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleUsernameChange(e) {
    this.setState({ username: e.target.value })
  }

  handlePasswordChange(e) {
    this.setState({ password: e.target.value })
  }

  handleSubmit(e) {
    e.preventDefault()
    this.props.submitLogIn(this.state.username, this.state.password)
  }

  render() {
    return (
      <div>
        <form onSubmit={ () => { this.handleSubmit; this.props.sendToLobby; }> 
          <label>
          Username:
            <input 
              type='text' 
              value={ this.state.username } 
              onChange={ this.handleUsernameChange }
            />
          </label>
          <label>
            Password:
            <input 
              type='text'
              value={ this.state.password }
              onChange={ this.handlePasswordChange }
            />
          </label>
          <input 
            type='submit' 
            value='Login'
          />        
        </form>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    submitLogIn: (username, password) => { dispatch(LogIn(username, password)) }
  }
}

export default connect(mapDispatchToProps)(LogInPage)