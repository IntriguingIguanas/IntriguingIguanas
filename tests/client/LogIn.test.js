import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import { LogInPage } from 'LogInPage'

describe('LogInPage', () => {
  it('should be able to run tests', () => {
	  expect(1 + 2).toEqual(3);
  });
  it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<LogInPage />, div);
  });
  it('should render login input forms', () => {
  	const loginPage = renderer.create(<LogInPage />).toJSON();
  	expect(loginPage).toMatchSnapshot();

  });
});
