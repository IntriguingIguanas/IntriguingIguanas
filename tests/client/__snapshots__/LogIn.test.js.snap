// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInPage should render login input forms 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      Username:
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label>
      Password:
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <input
      type="submit"
      value="Login"
    />
  </form>
</div>
`;
